lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2609,6 +2609,80 @@ LogicalResult ConvertAtenOp<AtenGeluBackwardOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenEmbeddingOp>::matchAndRewrite(
+    AtenEmbeddingOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  Value weight = adaptor.weight();
+  Value indices = adaptor.indices();
+  RankedTensorType outType =
+      typeConverter->convertType(op.getType()).cast<RankedTensorType>();
+
+  auto indicesType = indices.getType().dyn_cast<RankedTensorType>();
+  if (!indicesType || !indicesType.getElementType().isa<IntegerType>())
+    return op.emitError("Indices must be of integer tensor type");
+
+  if (indicesType.getRank() != 2)
+    return op.emitError("indices must be of rank 2");
+
+  auto weightType = weight.getType().cast<RankedTensorType>();
+  if (weightType.getRank() != 2)
+    return op.emitError("weight must be of rank 2");
+
+  // FIXME: padding_idx, scale_grad_by_freq and sparse are not handled yet.
```

Comment:
can you make sure to bail out on those cases?

Reply:
Done